Source/CronosPlayUnreal/Private/PlayCppSdkActor.cpp

```diff
@@ -23,36 +23,128 @@ using namespace com::crypto::game_sdk;
 
 ::com::crypto::game_sdk::WalletconnectClient *APlayCppSdkActor::_coreClient =
     NULL;
-APlayCppSdkActor *APlayCppSdkActor::_sdk = NULL;
 
 class UserWalletConnectCallback : public WalletConnectCallback {
+  private:
+    APlayCppSdkActor *PlayCppSdkPtr;
   public:
+    UserWalletConnectCallback(APlayCppSdkActor *PlayCppSdkPtr);
     void onConnected(const WalletConnectSessionInfo &sessioninfo) const;
     void onDisconnected(const WalletConnectSessionInfo &sessioninfo) const;
     void onConnecting(const WalletConnectSessionInfo &sessioninfo) const;
     void onUpdated(const WalletConnectSessionInfo &sessioninfo) const;
 };
 
-APlayCppSdkActor *APlayCppSdkActor::getInstance() { return _sdk; }
+UserWalletConnectCallback::UserWalletConnectCallback(
+    APlayCppSdkActor *PlayCppSdk)
+    : PlayCppSdkPtr(PlayCppSdk) {}
+
+void UserWalletConnectCallback::onConnected( // NOLINT : flase positive, virtual
+                                             // function cannot be static
+    const WalletConnectSessionInfo &sessioninfo) const {
+    UE_LOG(LogTemp, Log, TEXT("user c++ onConnected"));
+    if (PlayCppSdkPtr) {
+        APlayCppSdkActor *PlayCppSdk =
+            PlayCppSdkPtr; // we need it to create a new varialbe for lambda
+        // We could not Call PlayCppSdkPtr's function direclty, bcause it is not
+        // in game thread. Instead, we can create AsyncTask and pass the
+        // PlayCppSdkPtr and sessioninfo into it, then call the function in game
+        // thread.
+        AsyncTask(ENamedThreads::GameThread, [PlayCppSdk, sessioninfo]() {
```

Comment:
confirmed.
calling delegate from asynctask, which making caller side in unreal game thread